scripts/Stretch.py

```diff
@@ -0,0 +1,419 @@
+        already deposited at one or the other side of this segment.
+        If material was deposited at one side but not both,
+        the segment is moved into the direction of the deposited material,
+        to "push the wall"
+
+        Already deposited material is stored as segments.
+        vd1 is the array of the starting points of the segments
+        vd2 is the array of the ending points of the segments
+        For example, segment nr 8 starts at position self.vd1[8]
+        and ends at position self.vd2[8]
+        """
+        dist_palp = self.line_width / 2.0 # Palpation distance to seek for a wall
+        mrot = np.array([[0, -1], [1, 0]]) # Rotation matrix for a quarter turn
+        for i in range(len(orig_seq)):
+            ibeg = i # Index of the first point of the segment
+            iend = i + 1 # Index of the last point of the segment
+            if iend == len(orig_seq):
+                iend = i - 1
+            xperp = np.dot(mrot, orig_seq[iend] - orig_seq[ibeg])
+            xperp = xperp / np.sqrt((xperp ** 2).sum(-1))
+            testleft = orig_seq[ibeg] + xperp * dist_palp
+            materialleft = False # Is there already extruded material at the left of the segment
+            testright = orig_seq[ibeg] - xperp * dist_palp
+            materialright = False # Is there already extruded material at the right of the segment
+            if self.vd1.shape[0]:
+                relpos = np.clip(((testleft - self.vd1) * (self.vd2 - self.vd1)).sum(1)
+                                 / ((self.vd2 - self.vd1) * (self.vd2 - self.vd1)).sum(1), 0., 1.)
+                nearpoints = self.vd1 + relpos[:, np.newaxis] * (self.vd2 - self.vd1)
+                # nearpoints is the array of the nearest points of each segment
+                # from the point testleft
+                dist = ((testleft - nearpoints) * (testleft - nearpoints)).sum(1)
+                # dist is the array of the squares of the distances between testleft
+                # and each segment
+                if np.amin(dist) <= dist_palp * dist_palp:
+                    materialleft = True
+                # Now the same computation with the point testright at the other side of the
+                # current segment
+                relpos = np.clip(((testright - self.vd1) * (self.vd2 - self.vd1)).sum(1)
+                                 / ((self.vd2 - self.vd1) * (self.vd2 - self.vd1)).sum(1), 0., 1.)
+                nearpoints = self.vd1 + relpos[:, np.newaxis] * (self.vd2 - self.vd1)
+                dist = ((testright - nearpoints) * (testright - nearpoints)).sum(1)
+                if np.amin(dist) <= dist_palp * dist_palp:
+                    materialright = True
+            if materialleft and not materialright:
+                modif_seq[ibeg] = modif_seq[ibeg] + xperp * self.stretch
+            elif not materialleft and materialright:
+                modif_seq[ibeg] = modif_seq[ibeg] - xperp * self.stretch
```

Comment:
I Don't understand your question. This portion of code checks if, when extruding a segment of material, material was already deposited at the left, at the right, or both (when in a corridor).
If material is already present at the left of the segment, the first point of the segment is slightly moved to the left, in the direction of the wall, and the same for the right.
This operation is done for each point of the sequence.
The objective is to avoid gaps between lines during turns, easily observable without a form of compensation like this algorithm.